src/qtiCommonRenderer/renderers/interactions/GapMatchInteraction.js

```diff
@@ -385,15 +385,17 @@ var resetResponse = function(interaction) {
 };
 
 var _setPairs = function(interaction, pairs) {
-    _.each(pairs, function(pair) {
-        if (pair) {
-            setChoice(
-                interaction,
-                getChoice(interaction, pair[0]),
-                getGap(interaction, pair[1]).find('.gapmatch-content')
-            );
+    if ( _.isArray(pairs) && _.isArray(pairs[0])) {
+        _.each(pairs, function(pair) {
+            if (pair && pair[0] && pair[1]) {
+                setChoice(interaction, getChoice(interaction, pair[0]), getGap(interaction, pair[1]).find('.gapmatch-content'));
+            }
+        });
+    } else {
+        if (pairs && pairs[0] && pairs[1]) {
+            setChoice(interaction, getChoice(interaction, pairs[0]), getGap(interaction, pairs[1]).find('.gapmatch-content'));
```

Comment:
Isn't there a simple way to do it? (with some factorization)

This block is duplicated: 
```javascript
            if (pair && pair[0] && pair[1]) {
                setChoice(interaction, getChoice(interaction, pair[0]), getGap(interaction, pair[1]).find('.gapmatch-content'));
            }
```

`_setPairs` is called at a single place, in `setResponse`, line 415.

Maybe a better approach would be to apply the check inside `setResponse`, and then call `_setPairs` appropriately:
```javascript
var setResponse = function(interaction, response) {
    resetResponse(interaction);
    let pairs = pciResponse.unserialize(response, interaction);
    if ( _.isArray(pairs) && _.isArray(pairs[0])) {
        _.forEach(pairs, pair => _setPairs(interaction, pair));
    } else {
        _setPairs(interaction, pairs);
    }
};
```

Reply:
ys, it looks better